core/lib/vm_executor/src/oneshot/mod.rs

```diff
@@ -49,10 +53,52 @@ mod contracts;
 mod env;
 mod metrics;
 mod mock;
+#[cfg(test)]
+mod tests;
+
+/// Fast VM modes utilized for different kinds of operations.
+#[derive(Debug, Clone, Copy, Default)]
+pub struct OneshotExecutorVmModes {
+    /// Mode used for gas estimation.
+    pub gas_estimation: FastVmMode,
+    /// Mode used for `eth_call`s.
+    pub call: FastVmMode,
+    /// Mode used for transaction execution (doesn't include the validation step).
```

Comment:
Please also add the validation step so I can work on it.

Reply:
Validation will be covered by this param once it's supported by the fast VM; I'll make the doc comment clearer ("doesn't include the validation step for now since it's not supported by the fast VM").

Reply:
A bit of a chicken and egg problem 😄 I guess I will add that option in my PR implementing validation then.